src/ecc_bitcoin.c

```diff
@@ -104,22 +105,28 @@ void libsecp256k1_ecc_context_destroy(void)
 
 
 int libsecp256k1_ecc_sign_digest(const uint8_t *private_key, const uint8_t *data,
-                                 uint8_t *sig, ecc_curve_id curve)
+                                 uint8_t *sig, uint8_t *recid, ecc_curve_id curve)
 {
     (void)(curve);
-    secp256k1_ecdsa_signature signature;
+    secp256k1_ecdsa_recoverable_signature signature;
 
     if (!libsecp256k1_ctx) {
         libsecp256k1_ecc_context_init();
     }
 
-    if (secp256k1_ecdsa_sign(libsecp256k1_ctx, &signature, (const unsigned char *)data,
-                             (const unsigned char *)private_key, secp256k1_nonce_function_rfc6979, NULL)) {
-        int i;
+    if (secp256k1_ecdsa_sign_recoverable(libsecp256k1_ctx, &signature,
```

Comment:
The only difference in the functions is:
```
if (secp256k1_ecdsa_sig_sign(&ctx->ecmult_gen_ctx, &r, &s, &sec, &msg, &non, NULL)) {
...
secp256k1_ecdsa_signature_save(signature, &r, &s);
```
vs
```
int recid;
...
if (secp256k1_ecdsa_sig_sign(&ctx->ecmult_gen_ctx, &r, &s, &sec, &msg, &non, &recid)) {
...
secp256k1_ecdsa_recoverable_signature_save(signature, &r, &s, recid);
```
So each use the same `ecdsa_sig_sign` function. In that function, recid does not affect the signing process. In other words, an `if recid != NULL` would not change anything.

In general, I think it would be useful in the future to return the recid through our standard API and was thinking to add a commit to do so here.